### PR TITLE
Handle infinity as an invalid value

### LINKED ISSUE #5657: Plots should handle Infinity gracefully
#### Summary
Plots will currently fail to render a trace completely if it includes any infinite values. Infinite values can sometimes occur in telemetry where a calculation involves division by zero.

#### Expected vs Current Behavior
The easiest thing to do for now is probably treat it in the same way we do `NaN` - that is, don't plot it and instead show a data gap.

#### Steps to Reproduce
1. Create a Sine Wave Generator object
2. View the Sine Wave Generator in a plot
3. [Modify the Sine Wave Generator object to randomly return Infinity](https://github.com/nasa/openmct/compare/infinite-values)
4. Observe that no data is returned at all. Compare this to the case where `NaN` is returned and observe that the plot will render the real numbers, with gaps for the `NaN`s

#### Environment
* Open MCT Version: 2.0.8
* Deployment Type: local dev

#### Impact Check List

- [x] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by shefalijoshi**
Closes #5657

### Describe your changes:
Positive and Negative infinity are rejected as invalid values and not added to plots.
**Note**: Plotly seems to handle Infinity by not displaying the data - so no change it needed for those plots.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by unlikelyzero** [comment]
@shefalijoshi can you provide a means for us to test this by modifying the generator plugin to have an infinite value?

File: src/plugins/plot/configuration/PlotSeries.js [hunk 1]
```diff
@@ -394,7 +394,8 @@ export default class PlotSeries extends Model {
         const currentYVal = this.getYVal(point);
         const lastYVal = this.getYVal(data[insertIndex - 1]);
 
-        if (this.isValueInvalid(currentYVal) && this.isValueInvalid(lastYVal)) {
+        if (this.isValueInfinity(currentYVal)
```
Comment (akhenry) [inline src/plugins/plot/configuration/PlotSeries.js:397]: Based on order of operations this if statement will evaluate to true if one of these two cases is true:

* The currentYVal and lastYVal are invalid (including infinity)
OR 
* Just the currentYVal is infinity.

Why do we have this special case where the currentYVal is infinity? Why don't we also do this for other unplottable values?

**Review by akhenry — CHANGES_REQUESTED** [review]

File: src/plugins/plot/configuration/PlotSeries.js (hunk 1 above)
Comment (shefalijoshi) [reply src/plugins/plot/configuration/PlotSeries.js:397]: Honestly, this logic has been around for a while - I assumed it had something to do with data continuity. I can change it to fail for all unplottable values.

Comment (akhenry) [reply src/plugins/plot/configuration/PlotSeries.js:397]: The infinity logic is new though? We have a new OR statement which will shortcut the previous logic. Just wondering why?

**Comment by shefalijoshi** [comment]
This is awaiting discussion on how to handle unplottable values.

**Comment by akhenry** [comment]
@shefalijoshi I believe this PR is unblocked now?

**Comment by shefalijoshi** [comment]
Closing this PR as @khalidadil will create a new one with a fix.

**Pull request closed without merging**